### ISSUE TITLE
Walletconnect doesn't support multiple chains at once

### ISSUE THREAD

**Issue by ptsayli** [WalletConnect/walletconnect-monorepo#124]
We want to support multiple chains at once as being layer 2, but `sendCustomRequest` or `sendTransaction` removes `chainId` and send default `chainId`.

Let me know if you have any plans to support it soon.

**Comment by ptsayli** [comment]
In fact, walletconnect should be communication layer between two nodes. It shouldn't change the passed data at all. That's a bad practice as protocol. Then, there is no point of encryption if walletconnect SDK changes data at before encryption.

**Comment by pedrouid** [comment]
I think that supporting multiple chainIds is a great idea and I want the WalletConnect SDK to support it.

However regarding the "changing the passed data", the sendCustomRequest simply parses Ethereum JSON-RPC requests to be complaint with the spec. If you use the `eth_sendTransaction` method and use a different format then you shouldn't expect it to let it pass invalid formats.

Also keep in mind the difference between the WalletConnect protocol and the WalletConnect SDK. The WalletConnect SDK is follwoing the WalletConnect Protocol spec and the Ethereum JSON RPC API spec which are different things.

If you would like to make customRequests which are not Ethereum transactions then you should use a unique method (example: `matic_sendTransaction`) otherwise non-Matic wallets will not be able to support non-complaint requests

**Comment by pedrouid** [comment]
Also I have already proposed an EIP to include the chainId on the `eth_sendTransaction` but I won't change the SDK until this EIP passes to change the Ethereum JSON RPC API spec first

EIP-1637: ethereum/EIPs#1637

**Comment by pedrouid** [comment]
I've created the issue #125 to implement EIP-1637

**Comment by hsouf** [comment]
@pedrouid Can you please help ? 
I get "chain is not supported" when trying to connect using WalletConnect, here is my web3Modal config : 

```
const web3Modal = new Web3Modal({
  network: "arbitrum-rinkeby",
  cacheProvider: false,
  disableInjectedProvider: false,
  providerOptions:{ 
    walletconnect: {
      package: WalletConnectProvider,
      options: {
        rpc: {
          421611: "https://rinkeby.arbitrum.io/rpc"
        }
      }
}
},
});
```

**Comment by pedrouid** [comment]
This has been addressed on v2.0 protocol
